test/test.downscopedclient.ts

```diff
@@ -298,6 +343,18 @@ describe('DownscopedClient', () => {
       downscopedClient.setCredentials(credentials);
```

Comment:
You can test the expiration is calculated correctly (with the offset) by using a mock clock and run tick with expirationDelta - 1 and confirm the cached token is returned.